carbon-mb/components/transports/mqtt/src/main/java/org/wso2/carbon/andes/transports/mqtt/broker/PublisherAcknowledgementProcessor.java

```diff
@@ -114,8 +124,9 @@ public void ack(AndesMessageMetadata metadata) {
         if (qos == AbstractMessage.QOSType.EXACTLY_ONCE.ordinal()) {
```

Comment:
Fixed with f38fe28f8e006b16fcb36c7b922def8121f446e1. The value would be retrieved instead of taking the ordinal